src/apps/organisations/redux/actions.ts

```diff
@@ -102,12 +102,10 @@ const retrieveOrganisationAction = (orgUrl: string) => {
 
         const response = error.response;
 
-        let errorMsg = errorMsgResponse(response);
 
-        const errorMessage: string | undefined | {} | [] =
-          response?.data || response
-            ? STATUS_CODES_TO_MESSAGES[response.status] || errorMsg
-            : errorMsg;
+        const errorMsgResponse: {[key: string]: string} = {
+          "__detail__": "Action could not be completed. Please retry."
+        }
```

Comment:
I forgot about these bits. This should be replaced with something like:

```ts
let errorMsg = errorMsgResponse(response);

const errorMessage =
  response?.data || response
    ? STATUS_CODES_TO_MESSAGES[response.status] || errorMsg
    : errorMsg["__detail__"];
```

Reply:
Alright... that was a silly suggestion.

Instead, let's change `errorMsgResponse()` to look like this:

```ts
export function errorMsgResponse(response: AxiosResponse) {
  const errorMsgResponse: {[key: string]: string} = {
    "__detail__": "Action could not be completed. Please retry."
  };

  if (response.data && Object.prototype.hasOwnProperty.call(response.data, "detail")) { 
    errorMsgResponse["__detail__"] = response.data.detail;
  } else if (response.status in STATUS_CODES_TO_MESSAGES) {
    errorMsgResponse["__detail__"] = STATUS_CODES_TO_MESSAGES[response.status];
  }
  
  for (let key in response.data) {
    if (key === "__detail__") continue;
    errorMsgResponse[key] =
        Array.isArray(response.data[key])
          ? response.data[key].join(",")
          : response.data[key]
    } 
    
  return errorMsgResponse;
};
```

This also means hunting out the places where we call `errorMsgResponse()` and ensure that we cast the argument we're returning to an `AxiosResponse` (it should always be this), like so:

```ts
const response: AxiosResponse = error.response;
```

Reply:
Thanks @ibacher let me do it right a way

Reply:
@jwnasambu So the second part of this is to just eliminate anything that looks like this:

```
const errorMessage =
  response?.data || response
    ? STATUS_CODES_TO_MESSAGES[response.status] || errorMsg
    : errorMsg["__detail__"];
```

From the code base... actually anything that processes the `errMsg` after it's returned from `errorMsgResponse(Response)`. The idea here is that that processing will be done on the actual page itself.

Reply:
@ibacher thanks for the clear explanation though it seems I had eliminated that part already!

Reply:
@ibacher am sorry I have just seen that part and other proposed changes. Kindly let me fix and push the changes right away.